_meta/config/common.p2.yml.tmpl

```diff
@@ -32,6 +33,17 @@ inputs:
         - filesystem
         data_stream.dataset: system.filesystem
 
+  # Collecting log files
+  #- type: filestream
+  #  id: your-first-filestream-input
+  #  streams:
+  #    - #id: logfile-log.logs-566cd4b0-e5ce-11ed-bcd4-43d4f70535fd # is this id needed too?
```

Comment:
Yes you need this. The `id` on the input is only used by the agent to identify the input in the policy and is used for status reporting. It does nothing for filestream itself.

Each stream here results in a separate instance of the filestream input, where the filestream id is the stream level id you currently have commented out.

For a concrete example, take the configuration with two filestream inputs below:

```yaml
  - type: filestream
    id: your-first-filestream-input
    streams:
      - id: logfile-A
        data_stream:
          dataset: generic
        paths:
          - /var/log/*.log
      - id: logfile-A
        data_stream:
          dataset: generic
        paths:
          - /var/log/*.log
```

This generates two separate filestream inputs, which you can observe by looking in the `beat-rendered-config.yml` file in the agent diagnostics bundle for the `filestream-default` component. Note that `your-first-filestream-input` doesn't appear here.

```yaml
inputs:
    - data_stream:
        dataset: generic
      id: logfile-A
      index: logs-generic-default
      paths:
      - /var/log/*.log
      type: filestream
    - data_stream:
        dataset: generic
      id: logfile-B
      index: logs-generic-default
      paths:
      - /var/log/*.log
      type: filestream
```

This is definitely confusing, but it is because we have each stream resulting in a filestream input. We could force a one to one mapping between agent inputs and filestream inputs, but the consequence would be those inputs could only have one data stream as filestream has no way to label data based on the source it was ingested from. Maybe we need "document routing" at the Filebeat level as well.

Reply:
> Ok, so this means we can skip the id on the global level. My goal is for now to have the minimum required.

No unfortunately, Elastic Agent needs to the global input ID to track the input state and filestream needs the stream IDs to separate each instance of filestream in the registry. You need both IDs.

>  Somehow along the way streams became required which I think was accidental and not by design :-( I would also argue, we didn't make a great design choice here.

Agreed, I came to the same conclusion in the last part of my comment but didn't communicate it clearly enough. We are stuck with both input and stream IDs for now. There is probably an alternate syntax that doesn't require this.

Reply:
> No unfortunately, Elastic Agent needs to the global input ID to track the input state and filestream needs the stream IDs to separate each instance of filestream in the registry. You need both IDs.

Got it. It got confused because Fleet seems to generate the same id for both which should be fine, as they are used for different purposes. I'll re-add the other one too. If we merge it down to just the input, the same id can be used for both.